code/game/gamemodes/gang/gang.dm

```diff
@@ -262,23 +414,73 @@ GLOBAL_LIST_INIT(gang_outfit_pool, list(/obj/item/clothing/suit/jacket/leather,/
 	return gang_bosses
 
 /datum/game_mode/proc/shuttle_check()
+	posse_timer++
 	if(SSshuttle.emergencyNoRecall)
 		return
 	var/alive = 0
 	for(var/mob/living/L in GLOB.player_list)
 		if(L.stat != DEAD)
 			alive++
-
-	if((alive < (GLOB.joined_player_list.len * 0.4)) && ((SSshuttle.emergency.timeLeft(1) > (SSshuttle.emergencyCallTime * 0.4))))
-
+	if(alive < (GLOB.joined_player_list.len * 0.45 && (SSshuttle.emergency.mode == SHUTTLE_RECALL) || (SSshuttle.emergency.mode == SHUTTLE_IDLE) || (SSshuttle.emergency.timeLeft(1) > (SSshuttle.emergencyCallTime * 0.4))))
 		SSshuttle.emergencyNoRecall = TRUE
 		SSshuttle.emergency.request(null, set_coefficient = 0.4)
 		priority_announce("Catastrophic casualties detected: crisis shuttle protocols activated - jamming recall signals across all frequencies.")
+	if((alive < (GLOB.joined_player_list.len *  0.7)) && posse_timer >= 4)
+		posse_timer = 0
+		vigilante_vengeance()
+
+
+/datum/game_mode/proc/vigilante_vengeance()
+	var/list/mob/dead/observer/candidates = pollGhostCandidates("Would you like be a part of a Vigilante posse?", "pAI", null, FALSE, 100)
+	var/list/mob/dead/observer/finalists = list()
+	var/posse_size = 1+round(GLOB.joined_player_list.len * 0.06)+round(world.time/5000)
+	if(candidates.len)
+		for(var/n in 1 to min(candidates.len,posse_size))
+			finalists += pick_n_take(candidates)
+	else
+		message_admins("No ghosts were willing to join the posse")
+		posse_timer = 2
+		return
+	for(var/i in 1 to finalists.len)
+		var/mob/living/carbon/human/character = new(src)
+		var/equip = SSjob.EquipRank(character, "Assistant", 1)
+		character = equip
+		var/D
+		if(GLOB.latejoin.len)
+			D = get_turf(pick(GLOB.latejoin))
+		if(!D)
+			for(var/turf/T in get_area_turfs(/area/shuttle/arrival))
+				if(!T.density)
+					var/clear = 1
+					for(var/obj/O in T)
+						if(O.density)
+							clear = 0
+							break
+					if(clear)
+						D = T
+						continue
+		character.loc = D
```

Comment:
from line 449 to here. You'll probably want to use SSjob.SendToLateJoin once my PR is merged: #26647